source/isaaclab/test/assets/test_articulation.py

```diff
@@ -1509,6 +1514,95 @@ def test_write_root_state(self):
+            for device in ("cuda:0", "cpu"):
+                with self.subTest(num_articulations=num_articulations, device=device):
+                    with build_simulation_context(
+                        gravity_enabled=True, device=device, add_ground_plane=True, auto_add_lighting=True
+                    ) as sim:
+                        articulation_cfg = generate_articulation_cfg(articulation_type="single_joint")
+                        articulation, _ = generate_articulation(
+                            articulation_cfg=articulation_cfg, num_articulations=num_articulations, device=device
+                        )
+
+                        # Play the simulator
+                        sim.reset()
+                        # apply external force
+                        external_force_vector_b = torch.zeros(
+                            (num_articulations, articulation.num_bodies, 3), device=device
+                        )
+                        external_force_vector_b[:, 1, 1] = 10.0  # 10 N in Y direction
+                        external_torque_vector_b = torch.zeros(
+                            (num_articulations, articulation.num_bodies, 3), device=device
+                        )
+                        external_torque_vector_b[:, 1, 2] = 10.0  # 10 Nm in z direction
+
+                        # apply action to the articulation
+                        joint_pos = torch.ones_like(articulation.data.joint_pos) * 1.5708 / 2.0
+                        articulation.write_joint_state_to_sim(
+                            torch.ones_like(articulation.data.joint_pos), torch.zeros_like(articulation.data.joint_vel)
+                        )
+                        articulation.set_joint_position_target(joint_pos)
+                        articulation.write_data_to_sim()
+                        for _ in range(100):
+                            articulation.set_external_force_and_torque(
+                                forces=external_force_vector_b, torques=external_torque_vector_b
+                            )
+                            articulation.write_data_to_sim()
+                            # perform step
+                            sim.step()
+                            # update buffers
+                            articulation.update(sim.cfg.dt)
+
+                            # check shape
+                            self.assertEqual(
+                                articulation.data.body_joint_reaction_wrench_b.shape,
+                                (num_articulations, len(articulation.data.body_names), 6),
+                            )
+
+                        # calculate expected static
+                        mass = articulation.root_physx_view.get_masses()  # [num_env, num_bodies]
+                        pos_w, quat_w = articulation.root_physx_view.get_link_transforms().split([3, 4], dim=-1)
+                        quat_w = math_utils.convert_quat(quat_w, to="wxyz")
```

Comment:
Hm take these from articulation data?
```suggestion
                        # calculate expected static
                        mass = articulation.data.default_mass
                        pos_w, quat_w = articulation.data.body_state_w[:, :7].split([3, 4], dim=-1)
```